rs/system_api/tests/sandbox_safe_system_state.rs

```diff
@@ -349,6 +349,107 @@ fn mint_cycles_fails_caller_not_on_nns() {
     );
 }
 
+#[test]
```

Comment:
Could we please factor out the common part for the three new tests here: what varies if the input balance and amount to mint, the output are the minted cycles and balance after.